src/query/api/v1/handler/placement/add.go

```diff
@@ -115,41 +115,39 @@ func (h *AddHandler) Add(
 		return nil, err
 	}
 
-	service, err := Service(h.client, httpReq.Header)
+	service, algo, err := ServiceWithAlgo(h.client, httpReq.Header)
 	if err != nil {
 		return nil, err
 	}
 
-	if !req.Force {
-		curPlacement, _, err := service.Placement()
+	if req.Force {
+		newPlacement, _, err := service.AddInstances(instances)
 		if err != nil {
 			return nil, err
 		}
 
-		if err := validateAllAvailable(curPlacement); err != nil {
-			return nil, err
-		}
+		return newPlacement, nil
 	}
 
-	newPlacement, _, err := service.AddInstances(instances)
+	curPlacement, version, err := service.Placement()
 	if err != nil {
 		return nil, err
 	}
 
-	return newPlacement, nil
-}
+	if err := validateAllAvailable(curPlacement); err != nil {
```

Comment:
This should work, but I feel a more reusable approach would be passing a placement validation function to the service through options and call it there, in case one day we change the logic in placement service and forgot to update here, what do you think?

Reply:
That does sound more flexible, but so far this is the only use case we've had for this and I'd rather limit the scope in this PR to unblock progress.

Reply:
kk, can you make a TODO here?

Reply:
Opened https://github.com/m3db/m3cluster/issues/243